ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorizationContext.java

```diff
@@ -1103,6 +1103,9 @@ private VectorExpression getGenericUDFStructField(ExprNodeFieldDesc exprNodeFiel
   private int getStructFieldIndex(ExprNodeFieldDesc exprNodeFieldDesc) throws HiveException {
     ExprNodeDesc structNodeDesc = exprNodeFieldDesc.getDesc();
     String fieldName = exprNodeFieldDesc.getFieldName();
+    if (exprNodeFieldDesc.getIsList()) {
+      throw new HiveException("Could not vectorize expression: Vectorizing complex type LIST is not supported");
```

Comment:
The `Vectorizing complex type..` message is created by `getValidateDataTypeErrorMsg` during some validation type checks. Why aren't we hitting these checks before arriving into this method?

Reply:
We are hitting this message because we never do that check(call getValidateDataTypeErrorMsg) for the select clause expressions and do this check only for the group by expression, which is later in the operator tree for that particular vertex.